### PR TITLE
Localized date fns

### PR DESCRIPTION
**Pull request by tomazed**
I took a shot at doing localization with date-fns
I do not like the lines 
```
import { de as localeDE, enUS as localeENUS, es as localeES, fr as localeFR, hr as localeHR, it as localeIT, pl as localePL, zhCN as localeZNCN } from 'date-fns/locale'
let locale = {
  de: localeDE,
  enUS: localeENUS,
  es: localeES,
  fr: localeFR,
  hr: localeHR,
  it: localeIT,
  pl: localePL,
  znCN: localeZNCN
}
```
from https://github.com/advplyr/audiobookshelf/blob/master/client/plugins/init.client.js

but for the love of god, I could not simply do a `import locale from 'date-fns/locale'` just like stated in the [docs](https://date-fns.org/v2.29.3/docs/setDefaultOptions#examples)
There must be a better way, but this one works...

I also took the liberty to generate the months and weekdays through date-fns format function. I think this render 
https://github.com/advplyr/audiobookshelf/blob/5724887785b55a56c4e07505ea7c0d2384c0febf/client/strings/en-us.json#L630-L643
and
https://github.com/advplyr/audiobookshelf/blob/5724887785b55a56c4e07505ea7c0d2384c0febf/client/strings/en-us.json#L536-L547
useless

### CODE REVIEW

**Comment by tomazed** [comment]
having slept on in, just because it works does not mean it should be merged... there are better ways to do it, and I'm not convince putting ConfigDateFnsLocale in the strings was a good choice from a maintainabily point of view....

Let me find a better way

**Comment by tomazed** [comment]
It's better now, I learnt to use js import :sweat_smile:
@advplyr hope the code meet project coding guidelines. If you want some other clean up, telle me

File: client/plugins/i18n.js [hunk 1]
```diff
@@ -5,18 +5,18 @@ import { supplant } from './utils'
...
-  'pl': 'Polski',
-  'zh-cn': '简体中文 (Simplified Chinese)'
+  'de': { label: 'Deutsch', dateFnsLocale: 'de' },
+  'en-us': { label: 'English', dateFnsLocale: 'enUS' },
+  // 'es': { label: 'Español', dateFnsLocale: 'es' },
+  'fr': { label: 'Français', dateFnsLocale: 'fr' },
+  'hr': { label: 'Hrvatski', dateFnsLocale: 'hr' },
+  'it': { label: 'Italiano', dateFnsLocale: 'it' },
+  'pl': { label: 'Polski', dateFnsLocale: 'pl' },
+  'zh-cn': { label: '简体中文 (Simplified Chinese)', dateFnsLocale: 'znCN' },
```
Comment (advplyr) [inline client/plugins/i18n.js:15]:
```suggestion
  'zh-cn': { label: '简体中文 (Simplified Chinese)', dateFnsLocale: 'zhCN' },
```

**Comment by advplyr** [comment]
This is great, thanks for taking care of that! Tested and working for all languages.

**Pull request merged**